ores/scoring_systems/celery_queue.py

```diff
@@ -0,0 +1,287 @@
+        backend.store_result(body['id'], result=None, status="SENT")
+
+
+class CeleryQueue(ScoringSystem):
+
+    def __init__(self, *args, application, queue_maxsize=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.application = application
+        self.queue_maxsize = int(queue_maxsize) if queue_maxsize is not None \
+                             else None
+
+        self.redis = redis_from_url(self.application.conf.BROKER_URL)
+
+        if self.queue_maxsize is not None and self.redis is None:
+            logger.warning("No redis connection.  Can't check queue size")
+
+        expected_errors = (revscoring.errors.RevisionNotFound,
+                           revscoring.errors.PageNotFound,
+                           revscoring.errors.UserNotFound,
+                           revscoring.errors.DependencyError,
+                           mwapi.errors.RequestError,
+                           mwapi.errors.TimeoutError,
+                           TimeoutError)
+
+        @self.application.task(throws=expected_errors,
+                               queue=DEFAULT_CELERY_QUEUE,
+                               priority=0)
+        def _lookup_score_in_map(result_id, model_name):
+            logger.info("Looking up {0} in {1}".format(model_name, result_id))
+            score_map_result = self._process_score_map.AsyncResult(result_id)
+            try:
+                score_map = score_map_result.get(timeout=self.timeout)
+            except Exception as e:
+                logger.error(e)
+                raise
+            logger.info("Found up {0} in {1}!".format(model_name, result_id))
+            return score_map[model_name]
+
+        @self.application.task(throws=expected_errors,
+                               queue=DEFAULT_CELERY_QUEUE,
+                               priority=0)
+        def _process_score_map(
+             context_name, model_names, rev_id, root_cache=None,
+             injection_cache=None, include_features=False):
+            logger.info("Generating a score map for" +
+                        "{0}:{1}:{2}"
+                        .format(context_name, set(model_names), rev_id))
+            try:
+                score_map = ScoringSystem._process_score_map(
+                    self, context_name, model_names, rev_id,
+                    root_cache=root_cache,
+                    injection_cache=injection_cache,
+                    include_features=include_features)
+            except Exception as e:
+                logger.error(e)
+                raise
+            logger.info("Completed generating score map for" +
+                        "{0}:{1}:{2}"
+                        .format(context_name, set(model_names), rev_id))
+            return score_map
+
+        APPLICATIONS.append(application)
+
+        self._process_score_map = _process_score_map
+        self._lookup_score_in_map = _lookup_score_in_map
```

Comment:
There's a lot going on in `__init__` that could be clarified by splitting it out into separate functions